### DISCUSSION TITLE
RLS Policy not working as intended

### DISCUSSION THREAD

**Discussion by mall0xc** [supabase/supabase#21769] [Questions] [answered]
Hi all,

I have the following RLS policy that's doing my head in:

```
CREATE POLICY name
ON public."UsersToOrgs"
FOR INSERT
WITH CHECK (
    EXISTS (
        SELECT 1
        FROM
          "Organisations"
        WHERE
          id = organisation_id
          AND createdby = user_id
    )
);
```

The Organisations table has columns [id, name, createdby]
The UsersToOrgs table has columns [organisation_id, user_uid]

I want to allow a user to insert a row into the UsersToOrgs table if:

1. Theres a row in Organisations where id = NEW.organisation_id AND
2. That row has createdby = NEW.user_id

The above should work, but I keep getting forbidden errors, even with `returning="minimal"`

I can't seem to actually use NEW in my RLS policies, that was just added to better illustrate my point.

TLDR version is:
Allow the creator of an organisation to add themselves to it.

**Comment by GaryAustin1** [comment]
Does the user meet select RLS on organisation?
NEW does not work in RLS.  You just use the column names.

**Reply by mall0xc** [reply]
The user does not have select permissions, but note that I'm setting `returning="minimal"`. 

I understand I can't use NEW in my RLS policies, that was just added to better illustrate my point.

**Reply by GaryAustin1** [reply, marked as answer]
Your policy is on UserToOrgs for insert.  That does not need select RLS.
But in that policy you use 
```
SELECT 1
        FROM
          "Organisations"
```
That requires the user meet the RLS policy on Organisations.

**Reply by mall0xc** [reply]
That worked! It was my understanding that these RLS policies get executed by the role that created them (I swear I read that in the supabase docs somewhere).

I created the policy using the postgres role, which I thought would bypass RLS.

**Reply by GaryAustin1** [reply]
No RLS policies on all tables run as the user calling the initial table.  You can define 'security definer' functions will run as Postgres and use those in RLS to bypass RLS on other tables, which sometimes makes sense for performance reasons.
